### DISCUSSION TITLE
build.sh fails at STEP 14/18 [SOLVED]

### DISCUSSION THREAD

**Discussion by shark711** [fat-tire/resolve#54] [Q&A] [answered]
When trying to build with podman using build.sh, i get the following error:

STEP 14/18: RUN cd /tmp     && unzip *DaVinci_Resolve_Linux.zip     && if [ "`od -An -j 8 -N 6 -x --endian=big *DaVinci_Resolve_*_Linux.run`" = " 4149 0200 0000" ]; then             sed '0,/AI\x02/{s|AI\x02|\x00\x00\x00|}' -i *DaVinci_Resolve_*_Linux.run; fi     && ./*DaVinci_Resolve_*_Linux.run  --appimage-extract     && cd squashfs-root     && ./AppRun -i -a -y     && cat /tmp/squashfs-root/docs/License.txt     && rm -rf /tmp/*.run /tmp/squashfs-root /tmp/*.zip /tmp/*.pdf
Archive:  DaVinci_Resolve_Linux.zip
  inflating: DaVinci_Resolve_18.6.6_Linux.run   bad CRC 77fc7613  (should be ecc631bd)
  inflating: Linux_Installation_Instructions.pdf  
Error: building at STEP "RUN cd /tmp     && unzip *DaVinci_Resolve_Linux.zip     && if [ "`od -An -j 8 -N 6 -x --endian=big *DaVinci_Resolve_*_Linux.run`" = " 4149 0200 0000" ]; then             sed '0,/AI\x02/{s|AI\x02|\x00\x00\x00|}' -i *DaVinci_Resolve_*_Linux.run; fi     && ./*DaVinci_Resolve_*_Linux.run  --appimage-extract     && cd squashfs-root     && ./AppRun -i -a -y     && cat /tmp/squashfs-root/docs/License.txt     && rm -rf /tmp/*.run /tmp/squashfs-root /tmp/*.zip /tmp/*.pdf": while running runtime: exit status 2

Not sure if the "bad CRC" is causing the exit code (2) but i redownloaded the file, and also tried with latest v19 and get the bad CRC each time

$ podman images
REPOSITORY                    TAG         IMAGE ID      CREATED      SIZE
docker.io/library/rockylinux  8.6         8cf70153e062  2 years ago  202 MB

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04 LTS
Release:	24.04
Codename:	noble

$ glxinfo | grep "OpenGL version"
OpenGL version string: 4.6.0 NVIDIA 535.183.01

$ nvidia-smi
Tue Jul 16 20:42:27 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3080        Off | 00000000:2B:00.0  On |                  N/A |
|  0%   37C    P8              22W / 320W |    467MiB / 10240MiB |      6%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      2790      G   /usr/lib/xorg/Xorg                          206MiB |
|    0   N/A  N/A      3072      G   /usr/bin/gnome-shell                         61MiB |
|    0   N/A  N/A      4715      G   ...,262144 --variations-seed-version=1      104MiB |
|    0   N/A  N/A      6860      G   /usr/bin/gnome-text-editor                   10MiB |
|    0   N/A  N/A     11977      G   /snap/snap-store/1124/bin/snap-store         24MiB |
+---------------------------------------------------------------------------------------+

$ lscpu
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          43 bits physical, 48 bits virtual
  Byte Order:             Little Endian
CPU(s):                   32
  On-line CPU(s) list:    0-31
Vendor ID:                AuthenticAMD
  Model name:             AMD Ryzen 9 3950X 16-Core Processor
    CPU family:           23
    Model:                113
    Thread(s) per core:   2
    Core(s) per socket:   16
    Socket(s):            1
    Stepping:             0
    Frequency boost:      enabled
    CPU(s) scaling MHz:   48%
    CPU max MHz:          4761.2300
    CPU min MHz:          2200.0000
    BogoMIPS:             7000.17

Keen to hear if others had this issue.
Recently got rid of Windows and switch to linux, and unable to get this up and running

**Comment by fat-tire** [comment]
The line:

`inflating:bad CRC 77fc7613 (should be ecc631bd)`

Suggests that there is actually a corrupted .zip file.  CRC is a [cyclic redundancy check](https://en.wikipedia.org/wiki/Cyclic_redundancy_check), used to ensure a file does not contain any errors.  If a .zip fails a CRC check while uncompressing, it usually means something in the file is not what was expected.

If you like, you can use  `unzip -t <filename>.zip` to [verify the CRC](https://linux.die.net/man/1/unzip) outside of the build script (on the host side) to see if the file can in fact be uncompressed successfully.

I know you said you tried re-downloading the .zip, but are you 100% sure you replaced the older file with the newly downloaded one?  (And did you notice any difference, say in file size in the two downloads?) 

You can easily find if the downloaded files are identical or not:

Method 1:
```bash
$ diff file1 file2
Binary files file1 and file2 differ
```
(if they are the same, you will not get any output)

Method 2:

```bash
$ sha256sum file1
$ sha256sum file2
```
(each of the above commands will output a random-looking characters --a "hash" of the file. If the characters outputted are the same for each file, the file is the same. Otherwise, they differ.)

Also, instead of 18.6.6, maybe you want to try a different file-- say the latest beta for version 19, just to be 100% sure it's not that file itself.  (Maybe it is corrupted on Blackmagic's side?)

Let me know if any of these ideas helped!

**Comment by shark711** [comment]
Deleted all my downloaded zip files

Then
$ `cd ~/containers/resolve`
$ `wget https://swr.cloud.blackmagicdesign.com/DaVinciResolve/v19.0b5/DaVinci_Resolve_19.0b5_Linux.zip?verify=[REDACTED] -O DaVinci_Resolve_19.0b5_Linux.zip`

Checked for potential files with same name to be sure:
$ `sudo find / -iname "*davinci*.zip"`
/home/shark711/containers/resolve/DaVinci_Resolve_19.0b5_Linux.zip
find: ‘/run/user/1000/doc’: Permission denied
find: ‘/run/user/1000/gvfs’: Permission denied

$`unzip -t /home/shark711/containers/resolve/DaVinci_Resolve_19.0b5_Linux.zip`
Archive:  /home/shark711/containers/resolve/DaVinci_Resolve_19.0b5_Linux.zip
    testing: DaVinci_Resolve_19.0b5_Linux.run   OK
    testing: Linux_Installation_Instructions.pdf   OK
No errors detected in compressed data of /home/shark711/containers/resolve/DaVinci_Resolve_19.0b5_Linux.zip.


Deleted all my podman images `podman rmi -f $(podman images -q)` (only had one, but wanted to make sure)

ran build.sh again
Got similar error (different version this time, so different file)
STEP 14/18: RUN cd /tmp     && unzip *DaVinci_Resolve_Linux.zip     && if [ "`od -An -j 8 -N 6 -x --endian=big *DaVinci_Resolve_*_Linux.run`" = " 4149 0200 0000" ]; then             sed '0,/AI\x02/{s|AI\x02|\x00\x00\x00|}' -i *DaVinci_Resolve_*_Linux.run; fi     && ./*DaVinci_Resolve_*_Linux.run  --appimage-extract     && cd squashfs-root     && ./AppRun -i -a -y     && cat /tmp/squashfs-root/docs/License.txt     && rm -rf /tmp/*.run /tmp/squashfs-root /tmp/*.zip /tmp/*.pdf
Archive:  DaVinci_Resolve_Linux.zip
  inflating: DaVinci_Resolve_19.0b5_Linux.run   bad CRC 79b765c8  (should be 336201b6)
  inflating: Linux_Installation_Instructions.pdf  
Error: building at STEP "RUN cd /tmp     && unzip *DaVinci_Resolve_Linux.zip     && if [ "`od -An -j 8 -N 6 -x --endian=big *DaVinci_Resolve_*_Linux.run`" = " 4149 0200 0000" ]; then             sed '0,/AI\x02/{s|AI\x02|\x00\x00\x00|}' -i *DaVinci_Resolve_*_Linux.run; fi     && ./*DaVinci_Resolve_*_Linux.run  --appimage-extract     && cd squashfs-root     && ./AppRun -i -a -y     && cat /tmp/squashfs-root/docs/License.txt     && rm -rf /tmp/*.run /tmp/squashfs-root /tmp/*.zip /tmp/*.pdf": while running runtime: exit status 2


Funny thing is, when I ran it again after deleting the podman images, I got a different CRC number (without touching the zip file)
 inflating: DaVinci_Resolve_19.0b5_Linux.run   bad CRC `81538d94`  (should be 336201b6)
vs
  inflating: DaVinci_Resolve_19.0b5_Linux.run   bad CRC `79b765c8`  (should be 336201b6)

checked original file to be safe, and was still ok
~/containers/resolve$ `unzip -t DaVinci_Resolve_19.0b5_Linux.zip`
Archive:  DaVinci_Resolve_19.0b5_Linux.zip
    testing: DaVinci_Resolve_19.0b5_Linux.run   OK
    testing: Linux_Installation_Instructions.pdf   OK
No errors detected in compressed data of DaVinci_Resolve_19.0b5_Linux.zip.

Going to wipe my hard-drive and reinstall Ubuntu from scratch

**Comment by shark711** [comment]
sigh.....
I give up



Complete!
warning: /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID 8483c65d: NOKEY
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  325M  100  325M    0     0  48.9M      0  0:00:06  0:00:06 --:--:-- 52.7M
**Verifying archive integrity... Error in check sums 3600907162 3558731085**
Error: building at STEP "RUN if [[ `dnf list libxcrypt-compat` == *libxcrypt-compat* ]]; then export EXTRA_PACKS="${EXTRA_PACKS} libxcrypt-compat" ; fi        && if [[ `dnf list compat-openssl11` == *compat-openssl11* ]]; then export EXTRA_PACKS="${EXTRA_PACKS} compat-openssl11" ; fi        && if [[ `dnf list compat-openssl10` == *compat-openssl10* ]]; then export EXTRA_PACKS="${EXTRA_PACKS} compat-openssl10"; fi        && export NVIDIA_VERSION=$NVIDIA_VERSION        && export ARCH=$ARCH        && dnf update --refresh -y        && dnf install dnf-plugins-core xorg-x11-server-Xorg libXcursor unzip alsa-lib librsvg2 libGLU sudo module-init-tools libgomp xcb-util python39 -y        && dnf install epel-release -y && dnf install xcb-util-cursor -y        && if [ ! -z "${EXTRA_PACKS}" ]; then dnf install ${EXTRA_PACKS} -y ; fi        && if [[ "${NO_PIPEWIRE}" == 0 ]] ; then              dnf install libXi libXt libXtst procps dbus-x11 libSM pipewire libcurl-devel                  apr apr-util libXinerama libxkbcommon libxkbcommon-x11 libXrandr xcb-util-image xcb-util-keysyms xcb-util-renderutil xcb-util-wm                  libglvnd-opengl pulseaudio-libs nss libXcomposite libXdamage -y              && curl https://vault.centos.org/8-stream/AppStream/x86_64/os/Packages/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm -o /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm              && dnf -y remove pipewire-alsa              && rpm -i --nodeps --replacefiles /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm              && PINNEDSHA=`/usr/bin/sha256sum /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm`              && if [ "${PINNEDSHA}" != "a870db3bceeeba7f96a9f04265b8c8359629f0bb3066e68464e399d88001ae52  /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm" ]; then echo "bad checksum" ; exit 1; fi              && rm /tmp/alsa-plugins-pulseaudio-1.1.9-1.el8.x86_64.rpm ;           else              dnf install alsa-plugins-pulseaudio -y ;           fi        && curl https://us.download.nvidia.com/XFree86/Linux-${ARCH}/${NVIDIA_VERSION}/NVIDIA-Linux-${ARCH}-${NVIDIA_VERSION}.run -o /tmp/NVIDIA-Linux-${ARCH}-${NVIDIA_VERSION}.run        && bash /tmp/NVIDIA-Linux-${ARCH}-${NVIDIA_VERSION}.run --no-kernel-module --no-kernel-module-source --run-nvidia-xconfig --no-backup --no-questions --accept-license --ui=none        && rm -f /tmp/NVIDIA-Linux-${ARCH}-${NVIDIA_VERSION}.run        && rm -rf /var/cache/yum/*        && dnf remove -y epel-release dnf-plugins-core libcurl-devel        && dnf clean all": while running runtime: exit status 2
Build of resolve:19.0b5 is complete.  To run resolve, try:


if I run it again, i am back to same error
STEP 14/18: RUN cd /tmp     && unzip *DaVinci_Resolve_Linux.zip     && if [ "`od -An -j 8 -N 6 -x --endian=big *DaVinci_Resolve_*_Linux.run`" = " 4149 0200 0000" ]; then             sed '0,/AI\x02/{s|AI\x02|\x00\x00\x00|}' -i *DaVinci_Resolve_*_Linux.run; fi     && ./*DaVinci_Resolve_*_Linux.run  --appimage-extract     && cd squashfs-root     && ./AppRun -i -a -y     && cat /tmp/squashfs-root/docs/License.txt     && rm -rf /tmp/*.run /tmp/squashfs-root /tmp/*.zip /tmp/*.pdf
Archive:  DaVinci_Resolve_Linux.zip
  inflating: DaVinci_Resolve_19.0b5_Linux.run   bad CRC 78bb12c5  (should be 336201b6)
  inflating: Linux_Installation_Instructions.pdf 
  
  Keeping in mind this is a fresh install of ubuntu, so there is nothing installed other than the prerequisites listed

**Reply by shark711** [reply]
I have good news and bad news
Good news, after running it a 3rd time, it passed that step
Bad news is, it failed on step 18/18

STEP 18/18: CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve
COMMIT resolve:19.0b5
Getting image source signatures
Copying blob 879054335f94 skipped: already exists  
Copying blob bc70bd2f43a5 done   | 
Error: committing container for step {Env:[ARCH=x86_64 ZIPNAME=DaVinci_Resolve_19.0b5_Linux.zip EXTRA_PACKS= POWERTOOLS=powertools NVIDIA_VERSION=535.183.01 BUILD_X264_ENCODER_PLUGIN=0 USER=resolve USER_ID=1000 NO_PIPEWIRE=0 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Command:cmd Args:[__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve] Flags:[] Attrs:map[] Message:CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve Original:CMD __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia __VK_LAYER_NV_optimus=NVIDIA_only /opt/resolve/bin/resolve}: copying layers and metadata for container "aacd196888932bf3e68bd4edbff206b959ac1ab3669775f14b271d34987454d3": writing blob: storing blob to file "/var/tmp/container_images_storage37436088/1": happened during read: Digest did not match, expected sha256:bc70bd2f43a50b2feccabf7cf7bb5740c8469b4a2bdd954e181ff54dc44ecd26, got sha256:b3542456d20ef625daa9e7bc555157e6bd0fcb9566b7144020cd7b237119f04f


... can only assume my Samsung SSD 970 EVO Plus 2TB has issues, cos i don't understand what else this could be

**Comment by fat-tire** [comment]
Yes unfortunately with the inconsistency of these errors, and the fact that they seem to be related to checksums and verifying data, it looks like you may be having SSD issues :(  At the very least, it may be a good idea to back up your most important data ASAP, then maybe replace the drive.  Also, you might check the `dmesg` command (requires `sudo` on recent versions of Ubuntu) and look for any i/o errors or other warnings...

**Comment by shark711** [comment]
Yeah we are going beyond the realm of this container and i appreciate all the help you have provided @fat-tire 

I don't expect you to solve this, but for what it is worth
All the disk checks came back with good results, including `sudo fsck -cc /dev/nvme0n1p1`, `sudo badblocks -v /dev/nvme0n1`, smartctl (long and short tests). Took a while for them all to complete

$ `sudo badblocks -v /dev/nvme0n1`
Checking blocks 0 to 1953514583
Checking for bad blocks (read-only test): done                                                 
Pass completed, 0 bad blocks found. (0/0/0 errors)

$ `sudo smartctl -a /dev/nvme0n1`
...
Num  Test_Description  Status                       Power_on_Hours  Failing_LBA  NSID Seg SCT Code
 0   Short             Completed without error               11944            -     -   -   -    -
 1   Extended          Completed without error               11944            -     -   -   -    -

the ONLY thing i found was
$ `sudo smartctl -a /dev/nvme0n1`
...
Error Information (NVMe Log 0x01, 16 of 64 entries)
Num   ErrCount  SQId   CmdId  Status  PELoc          LBA  NSID    VS  Message
  0       5411     0  0x0008  0x4004      -            0     0     -  **Invalid Field in Command**
...
 
And this doesn't appear to be a disk issue, not sure if this is perhaps firmware/driver issue

**Comment by fat-tire** [comment]
That's good that you're not seeing too many errors... I don't know what that SMART error means actually.  SMART was popular with hdd, but I don't see it referenced so much with ssd lately.  Did you keep an eye on `dmesg` when you did experience the checksum errors?  (You can follow it "live" with `sudo dmesg -w`) Often if you see i/o errors there you'll know something is wrong.

Also, did you try a `fsck -fy /dev/nvme0n1` as well as the `-cc` option?  cc wouldn't make any changes or attempt at repairs, while fy would... but if you're not seeing any errors to fix maybe it wouldn't make a difference.

Anyway, hope you finally got the container working :)

**Reply by shark711** [reply, marked as answer]
I found the problem. One of my memory modules are bad

https://x.com/sh4rk711/status/1814226197586522269

Explains the inconsistencies

your recommendation on `dmesg` started to list segmentation faults on random commands, such as `apt-get update` etc. which made me look at memory, thanks for that @fat-tire

**Comment by shark711** [comment]
Yip that was it... thanks for your help @fat-tire 
![image](https://github.com/user-attachments/assets/83b58b60-76df-47b9-a8de-6ae0506f79a7)

Complains about it not supporting CUDA 12.3, but after manually selecting openCL and my GPU it starts up

$ `./resolve.sh nvidia-smi`
non-network local connections being added to access control list
------------------------------------------------------------------------------
No "container-machine-id" file found for this container.
Creating new one derived from host machine-id (94c36052f6d346f091bf819b7aaaaf08)...
This container's machine-id file can now be found at /home/shark711/containers/resolve/mounts/container-machine-id and may be replaced if needed.
The container's /etc/machine-id  : 8f3d6233f82cff65b7643e94392c8225
The network driver setting is          : --network=none
Bind-mounted directories of interest   :
  CONTAINER (CentOS 8)                  -> HOST (Ubuntu)
  ----------------------------------------------------------------------------
  /home/resolve                         -> /home/shark711/containers/resolve/mounts/resolve-home
  /opt/resolve/logs                     -> /home/shark711/containers/resolve/mounts/logs
  /opt/resolve/configs                  -> /home/shark711/containers/resolve/mounts/configs
  /opt/resolve/easyDCP                  -> /home/shark711/containers/resolve/mounts/easyDCP
  /opt/resolve/.license                 -> /home/shark711/containers/resolve/mounts/license
  /opt/resolve/'Resolve Disk Database'  -> /home/shark711/containers/resolve/mounts/database
  /var/BlackmagicDesign/DaVinci Resolve -> /home/shark711/containers/resolve/mounts/BlackmagicDesign
  /opt/resolve/Media                    -> /home/shark711/containers/resolve/mounts/Media
Fri Jul 19 21:09:38 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3080        Off | 00000000:2B:00.0  On |                  N/A |
|  0%   36C    P8              19W / 320W |    489MiB / 10240MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
+---------------------------------------------------------------------------------------+

**Comment by fat-tire** [comment]
Terrific!  I mean, not that you had bad RAM, but that you figured the issue!
